src/urllib3/util/connection.py

```diff
@@ -1,12 +1,12 @@
 import socket
-from typing import List, Optional, Tuple, Union
+from typing import Optional, Sequence, Tuple, Union
 
 from urllib3.exceptions import LocationParseError
 
 from .wait import wait_for_read
 
 SOCKET_GLOBAL_DEFAULT_TIMEOUT = socket._GLOBAL_DEFAULT_TIMEOUT  # type: ignore
-SocketOptions = List[Tuple[int, int, Union[int, bytes]]]
+SocketOptions = Sequence[Tuple[int, int, Union[int, bytes]]]
```

Comment:
Since we're changing this to `Sequence` can we add a small test that makes sure sending for instance a tuple of socket options works okay?

Reply:
I changed the `test_socket_options` test and add a tuple of socket options